drag_and_drop_v2/drag_and_drop_v2.py

```diff
@@ -353,14 +352,18 @@ def do_attempt(self, data, suffix=''):
         self._mark_complete_and_publish_grade()  # must happen before _get_feedback
 
         overall_feedback_msgs, misplaced_ids = self._get_feedback()
+        misplaced_items = [self._get_item_definition(int(item_id)) for item_id in misplaced_ids]
```

Comment:
Is the int() required around item_id?

Also could you combine this and the loop on line 358 into one loop?
You are looping through misplaced_items 2 times.